Source/items.cpp

```diff
@@ -1334,387 +1334,310 @@ int PLVal(int pv, int p1, int p2, int minv, int maxv)
-		v12 = &item[v8]._iPLMag;
-		goto LABEL_62;
+		item[i]._iPLMag -= r;
+		break;
 	case IPL_DEX:
-		v11 = &item[v8]._iPLDex;
-		goto LABEL_115;
+		item[i]._iPLDex += r;
+		break;
 	case IPL_DEX_CURSE:
-		v12 = &item[v8]._iPLDex;
-		goto LABEL_62;
+		item[i]._iPLDex -= r;
+		break;
 	case IPL_VIT:
-		v11 = &item[v8]._iPLVit;
-		goto LABEL_115;
+		item[i]._iPLVit += r;
+		break;
 	case IPL_VIT_CURSE:
-		v12 = &item[v8]._iPLVit;
-		goto LABEL_62;
+		item[i]._iPLVit -= r;
+		break;
 	case IPL_ATTRIBS:
-		v26 = v8;
-		item[v26]._iPLStr += v10;
-		item[v26]._iPLMag += v10;
-		item[v26]._iPLDex += v10;
-		item[v26]._iPLVit += v10;
+		item[i]._iPLStr += r;
+		item[i]._iPLMag += r;
+		item[i]._iPLDex += r;
+		item[i]._iPLVit += r;
 		break;
 	case IPL_ATTRIBS_CURSE:
-		v27 = v8;
-		item[v27]._iPLStr -= v10;
-		item[v27]._iPLMag -= v10;
-		item[v27]._iPLDex -= v10;
-		item[v27]._iPLVit -= v10;
+		item[i]._iPLStr -= r;
+		item[i]._iPLMag -= r;
+		item[i]._iPLDex -= r;
+		item[i]._iPLVit -= r;
 		break;
-	case IPL_GETHIT_CURSE:
-		v11 = &item[v8]._iPLGetHit;
-		goto LABEL_115;
 	case IPL_GETHIT:
-		v12 = &item[v8]._iPLGetHit;
-		goto LABEL_62;
+		item[i]._iPLGetHit += r;
+		break;
+	case IPL_GETHIT_CURSE:
+		item[i]._iPLGetHit -= r;
+		break;
 	case IPL_LIFE:
-		v28 = v9 << 6;
-		v29 = &item[v8]._iPLHP;
-		goto LABEL_73;
+		item[i]._iPLHP += r << 6;
+		break;
 	case IPL_LIFE_CURSE:
-		v30 = v9 << 6;
-		v31 = &item[v8]._iPLHP;
-		goto LABEL_75;
+		item[i]._iPLHP -= r << 6;
+		break;
 	case IPL_MANA:
-		item[v8]._iPLMana += v9 << 6;
-		goto LABEL_92;
+		item[i]._iPLMana += r << 6;
+		drawmanaflag = TRUE;
+		break;
 	case IPL_MANA_CURSE:
-		item[v8]._iPLMana -= v9 << 6;
-		goto LABEL_92;
+		item[i]._iPLMana -= r << 6;
+		drawmanaflag = TRUE;
+		break;
 	case IPL_DUR:
-		v32 = v8;
-		v33 = item[v8]._iMaxDur;
-		v34 = v33;
-		v35 = v10 * v33 / 100;
-		item[v32]._iDurability += v35;
-		item[v32]._iMaxDur = v35 + v34;
+		r2 = r * item[i]._iMaxDur / 100;
+		item[i]._iMaxDur += r2;
+		item[i]._iDurability += r2;
 		break;
 	case IPL_DUR_CURSE:
-		v36 = v8;
-		v37 = item[v8]._iMaxDur - v9 * item[v8]._iMaxDur / 100;
-		item[v8]._iMaxDur = v37;
-		if (v37 < 1)
-			item[v36]._iMaxDur = 1;
-		item[v36]._iDurability = item[v36]._iMaxDur;
+		item[i]._iMaxDur -= r * item[i]._iMaxDur / 100;
+		if (item[i]._iMaxDur < 1)
+			item[i]._iMaxDur = 1;
+		item[i]._iDurability = item[i]._iMaxDur;
 		break;
 	case IPL_INDESTRUCTIBLE:
-		v38 = DUR_INDESTRUCTIBLE;
-		goto LABEL_119;
+		item[i]._iDurability = 255;
+		item[i]._iMaxDur = 255;
+		break;
 	case IPL_LIGHT:
-		v28 = param1;
-		v29 = &item[v8]._iPLLight;
-	LABEL_73:
-		*v29 += v28;
+		item[i]._iPLLight += param1;
 		break;
 	case IPL_LIGHT_CURSE:
-		v30 = param1;
-		v31 = &item[v8]._iPLLight;
-	LABEL_75:
-		*v31 -= v30;
+		item[i]._iPLLight -= param1;
 		break;
 	case IPL_FIRE_ARROWS:
-		v24 = v8;
-		item[v24]._iFlags |= 8u;
-	LABEL_77:
-		item[v24]._iFMinDam = param1;
-		item[v24]._iFMaxDam = param2;
+		item[i]._iFlags |= 8;

```

Comment:
_iFlags is the item_special_effect enum
```suggestion
		item[i]._iFlags |= ISPL_FIRE_ARROWS;
```